### DISCUSSION TITLE
Display area in more-info dialogs

### DISCUSSION THREAD

**Discussion by SeanPM5** [home-assistant/frontend#11174] [Other feature requests]
**Home Assistant release with the issue:**
0.98.3

**Description of problem:**
More info dialogs should display which Area a device belongs to.

A user could have two or more devices with identical friendly names. This is probably especially common with lights, where you could have a "Floor Lamp" in the office area and an identically named "Floor Lamp" in the living room area for example. There's currently no way to differentiate these since it does not display area on the more-info dialog.

Here is an idea of where it could be placed:

![example](https://i.imgur.com/7iCbH4y.jpg)

(This was just a quick example and doesn't represent the styling, it obviously wouldn't have red text or anything.)

**Comment by SeanPM5** [comment]
Also related to this, allow the user to assign an area from the edit cog on this screen. If the user sees that their device isn't assigned to an area they would be able to easily do it here, without having to head to the Integrations section.

![editarea](https://i.imgur.com/wCjJiOk.jpg)

**Comment by balloob** [comment]
So this brings us to a bit bigger topic:

Areas are attached to devices. A device can have multiple entities linked to it.

Would it be confusing if we conflate them here, or is that the right thing to do? Should we also show the other related entities ? If we have a device tracker and a battery sensor for this device, should we show battery level and if it's online ?

**Reply by amitkeret** [reply]
I am very much in support of this idea.
This is also related to my [feature request](https://github.com/home-assistant/frontend/discussions/11070) to have the area as available "secondary-info".

**Comment by iantrich** [comment]
I'm all for showing of the location in the more-info, that makes a lot of sense.

As for editing, I like the Area edit being attached to the entity edit screen and the preference is then fed up to the device. I don't see that being confusing.

**Comment by iantrich** [comment]
@balloob is the only way to get the area to get the full device registry from the backend?

**Comment by balloob** [comment]
@iantrich yeah, but that's fine, it's probably already loaded.

**Comment by iantrich** [comment]
As in that we could pass it along to the dialog? I don't quite follow

**Comment by balloob** [comment]
Our `subscribeX` methods cache the results, so when you request it, it already has the data.

**Comment by iantrich** [comment]
ah, cool. I did not know that. Thanks.

**Comment by spacegaier** [comment]
@SeanPM5 I like the idea. Material Design however does not foresee two titles in the header. Plus it feels out of whack if the main title is not in one line with the header icons or is that only me?

I also thought about making that a link to the area config page, but there are two cons:
1. Visually the area name needs to stay subdued to fit in, but it would be inconsistent for it to be a link then and not look like one.
2. Clicking on the header bar widens any dialog and that click/press could then accidentally hit the area name and navigate.

![image](https://user-images.githubusercontent.com/114137/105741195-50024500-5f3a-11eb-8577-22d6f5aedee6.png)

**Comment by SeanPM5** [comment]
That looks fine to me personally and I usually have OCD for those tiny little details. I feel like most people won't notice/care and will just appreciate the area info being there. But would be interested to hear a second opinion.

I think it's best to avoid making it a link because of what you mentioned.
